dump_queue_to_files.py

```diff
@@ -0,0 +1,76 @@
+import argparse
+import functools
+import uuid
+from pathlib import Path
+from typing import Collection
+
+from google.cloud import storage
+
+from rabbit_context import RabbitContext
+
+
+def start_listening_to_rabbit_queue(queue, on_message_callback, timeout=30):
+    rabbit = RabbitContext(queue_name=queue)
+    connection = rabbit.open_connection()
+
+    connection.call_later(
+        delay=timeout,
+        callback=functools.partial(_timeout_callback, rabbit))
+    rabbit.channel.basic_consume(
+        queue=queue,
+        on_message_callback=on_message_callback)
+    rabbit.channel.start_consuming()
+
+
+def _timeout_callback(rabbit):
+    print('Message queue appears to be drained')
+    rabbit.close_connection()
+
+
+def dump_messages(queue_name, output_file_path):
+    file_paths = []
+    start_listening_to_rabbit_queue(queue_name,
+                                    functools.partial(_rabbit_message_received_callback,
+                                                      output_file_path=output_file_path, file_paths=file_paths))
+    return file_paths
+
+
+def _rabbit_message_received_callback(ch, method, _properties, body, output_file_path, file_paths):
+    output_file = output_file_path.joinpath(f'{str(uuid.uuid4())}.json')
+    file_paths.append(output_file)
+    dump_message_to_file(output_file,  body)
+    ch.basic_ack(delivery_tag=method.delivery_tag)
+
+
+def dump_message_to_file(print_file_path: Path, message_body):
+    with open(print_file_path, 'w') as print_file:
+        print_file.write(message_body.decode("utf-8"))
+
+
+def copy_files_to_gcs(file_paths: Collection[Path], queue_name):
```

Comment:
Copying 60000 odd individual files to GCS slows down to a crawl of 50ish per second, I believe due to rate limits on number of uploads. If we zip the files up into a single archive it ends up as one file 30MB file which uploads in seconds.

Reply:
I like your zip solution